File: java/mysql/Create.java
//jar@http://central.maven.org/maven2/mysql/mysql-connector-java/6.0.5/mysql-connector-java-6.0.5.jar  

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;  

public class Create { 	 	
  	  	 
    public String testparams = "{\"hostname\":\"us-cdbr-sl-dfw-01.cleardb.net\","
  	 		+ "\"username\":\"b319ac7e3853e4\","
  	 		+ "\"password\":\"db586ae5\","
  	 		+ "\"dbname\":\"ibmx_b20a18ec42f7a48\","
  	 		+ "\"tablename\":\"student_username\","
  	 		+ "\"studentFirstName\":\"John\","
  	 		+ "\"studentLastName\":\"Doe\","
  	 		+ "\"studentId\":\"12345\"}"; 	  	 
    
    public static void main(String[] args) { 	   
    	Create hello = new Create(); 	   
	    System.out.println(main(new JsonParser().parse(hello.testparams).getAsJsonObject()));     
    } 	  	 
    
    public static JsonObject main(JsonObject args) { 	   
        
    	 JsonObject output = new JsonObject();
    	 
    	 String hostname = args.getAsJsonPrimitive("hostname").getAsString().isEmpty()?"SANDBOX_MYSQL_HOST":args.getAsJsonPrimitive("hostname").getAsString();
    	 String dbname = args.getAsJsonPrimitive("dbname").getAsString().isEmpty()?"SANDBOX_MYSQL_DATABASE":args.getAsJsonPrimitive("dbname").getAsString();
    	 String username = args.getAsJsonPrimitive("username").getAsString().isEmpty()?"SANDBOX_MYSQL_USER":args.getAsJsonPrimitive("username").getAsString();
    	 String password = args.getAsJsonPrimitive("password").getAsString().isEmpty()?"SANDBOX_MYSQL_PASSWORD":args.getAsJsonPrimitive("password").getAsString();
    	 
    	 String jdbcUrl = "jdbc:mysql://" + hostname + "/" + dbname + "?user=" + username + "&password=" + password + "&useLegacyDatetimeCode=false&serverTimezone=America/Chicago";
    	 try {
				  Class.forName("com.mysql.cj.jdbc.Driver");
	 			  Connection conn = DriverManager.getConnection(jdbcUrl);
	 		  
		 			String query = "INSERT INTO " + args.getAsJsonPrimitive("tablename").getAsString() + " (firstname, lastname, student_id)"
		    			+ " values (?, ?, ?)";
		    	PreparedStatement stat = conn.prepareStatement(query);
		    	stat.setString(1, args.getAsJsonPrimitive("studentFirstName").getAsString());
		    	stat.setString(2, args.getAsJsonPrimitive("studentLastName").getAsString());
		    	stat.setString(3, args.getAsJsonPrimitive("studentId").getAsString());
	    	
	     		stat.execute();
	     		output.addProperty("status", "Success Created");
	     		
		   		stat.close();
		  		conn.close();
			} catch (ClassNotFoundException e) {
				output.addProperty("err", e.getMessage());
			} catch (SQLException e) {
				output.addProperty("err", e.getMessage());
			}
	 		 
	 	  
   		 return output;	 
    } 	
}

File: nodejs/mysql/create.js
var testparams = {
    "hostname": "", 
    "username": "",
    "password": "",
    "dbname": "",
    "tablename": "student_username",
    "student_first_name": "John",
    "student_last_name": "Doe",
    "student_id": "12345"
};

var mysql = require('mysql');

function main(params) {

    var dbConfig = {
            host     : params.hostname || "SANDBOX_MYSQL_HOST",
            user     : params.username || "SANDBOX_MYSQL_USER",
            password : params.password || "SANDBOX_MYSQL_PASSWORD",
            database : params.dbname || "SANDBOX_MYSQL_DATABASE",
            connectionLimit : 5,
    };
    var pool  = mysql.createPool(dbConfig);

    var query = "INSERT INTO " + params.tablename + " SET ?";
    var values = {
		"firstname"  : params.student_first_name,
		"lastname"   : params.student_last_name,
		"student_id" : params.student_id
	}

    return new Promise(function(resolve, reject) {
        pool.query(query, values, function (err, data, fields) {
            pool.end();
            if (err) return reject({err});
            
            return resolve({data});
        });	
    })
}

if (require.main === module) {
    main(testparams).then(function(result) {
        console.log(JSON.stringify(result));
    })
    .catch(function (err) {
        console.error(err);
    });
}

exports.main = main;